ps_deployment/README.md

```diff
@@ -0,0 +1,119 @@
+		
+		* pros:
+			* separating code and image development - probably advantage in bigger projects where development and devops is separated
+		* cons:
+			* not copying .git to image, using default paths (/opt/app/src), maybe others which can probably require our scripts rework
+			
+	* normal image build (eg. `docker build ...`):
+		* pros:
+			* already in use in our environment
+			* no need to implement additional tool
+			* clean approach
+
+| ---------------------------------- | ----------- | ------------ |
+| image name                         | build time  | size         |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_github_base              | 4m42.531s   | 797MB        |
+| ps_normal_github_final             | 0m12.925s   | 797MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_normal_rpm_base                 | 5m8.026s    | 804MB        |
+| ps_normal_rpm_final                | 0m15.766s   | 804MB        |
+| ---------------------------------- | ----------- | ------------ |
+| ps_s2i_github_base                 | 4m38.620s   | 1.12GB       |
+| ps_s2i_github_final                | 0m10.715s   | 1.12GB       |
+| ---------------------------------- | ----------- | ------------ |
+
+### High level build flow strategies
+
+OpenShift vs. Public service
+	* OpenShift:
+		* pros:
+			* flexible
+			* no waiting in queue
+		* cons:
+			* additional maintenance
+			* additional resources = additional costs/failures because of resource limit
+	
+	* Public service:
+		* pros:
+			* our current approach - no radical changes required
+			* simple and straightforward
+			* free
+			
+		* cons:
+			* dependency on external service
+			* long queue time in case of service issue/high load
+
+#### OpenShift
+	* Source-to-Image -
+		 Source-to-Image (`sourceStrategy` in OC configuration) is using s2i tool for image deployment
+	
+	Links:
+	* [s2i details](https://docs.openshift.com/container-platform/3.11/architecture/core_concepts/builds_and_image_streams.html#source-build)
+	* [s2i strategy options](https://docs.openshift.com/container-platform/3.11/dev_guide/builds/build_strategies.html#source-to-image-strategy-options)
+	* [s2i tool blog](https://www.openshift.com/blog/create-s2i-builder-image)
+
+	* Image Stream - current configuration
+	* Custom Build - need more details
+
+## Conclusion
+
+	* Source-to-image (S2I) - no performance improvements, bigger image
+	* RPM vs github:
+		* from performance perspective basically same results 
+		* github repos should not have issues with OS version - will make deployment easier
```

Comment:
We had those problems in the past...

And we will have different dependencies for CLI and for service.

Alternative solution mixing those two:
- Installing from copr-repo that is built from `master`/`stable` branch. (I set up that last week.)
- :heavy_plus_sign: rpm versions for everything
- :heavy_plus_sign: no waiting for Fedora process 
- :heavy_minus_sign: We can't react to new commit to branch -- the build is not finished in that time. (Yes, this can be a problem.)

Reply:
please check https://github.com/packit-service/research/pull/52#discussion_r448282810 - there is discussed same topic